### DISCUSSION TITLE
How is the depth map resized after decimation filter?

### DISCUSSION THREAD

**Discussion by charlielito** [realsenseai/realsense-ros#3081] [Q&A] [answered]
According with https://github.com/IntelRealSense/librealsense/blob/master/doc/post-processing-filters.md#decimation-filter decimation filter downsamples the depth map depending on the parameters (also by inspecting the code https://github.com/IntelRealSense/librealsense/blob/4673a37d981164af8eeb8e296e430fc1427e008d/src/proc/decimation-filter.cpp#L204 it can be confirmed that the output is downsampled). Nonetheless, the published depth map image corresponds with the RGB image resolution. 
Does anyone know where and what type of upsampling is then made to the depth map to match the RGB image?

**Comment by Nir-Az** [comment, marked as answer]
Hi @charlielito ,
this example documentation might explain it.
https://github.com/IntelRealSense/librealsense/tree/development/examples/align

hope it helps :)

**Reply by charlielito** [reply]
Thanks! So the answer is `The interpolation used needs to be of type  Nearest Neighbor to avoid introducing non-existing values`
